fds.go

```diff
@@ -248,20 +255,21 @@ func (f *Fds) Listener(id string) (net.Listener, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 
-	return f.listenerLocked(id)
+	ln, _, err := f.listenerLocked(id)
+	return ln, err
 }
 
-func (f *Fds) listenerLocked(id string) (net.Listener, error) {
+func (f *Fds) listenerLocked(id string) (net.Listener, *fd, error) {
```

Comment:
suggestion: instead of returning the fd here, why not have this function mark `fd.used = true`?

Reply:
Indeed, that seems to be the same, but simpler!
Done!